core/src/replay_stage.rs

```diff
@@ -697,6 +697,7 @@ impl ReplayStage {
             let unfrozen_gossip_verified_vote_hashes = UnfrozenGossipVerifiedVoteHashes::default();
             let mut latest_validator_votes_for_frozen_banks =
                 LatestValidatorVotesForFrozenBanks::default();
+            let mut vote_slots = HashSet::default();
```

Comment:
the only downside of hoisting all the way up here is that it's possible we get some very far distance between current slot and root (practically speaking, this will likely hit 256 or so for most nodes after experiencing some fork/lockout scenario). I think that's probably fine.. but wondering if we should ever resize this down or something